Descent3/init.cpp

```diff
@@ -1406,16 +1404,30 @@ void InitIOSystems(bool editor) {
       Error("Failed to get executable path\n");
     } else {
       std::filesystem::path executablePath(exec_path);
-      std::string baseDirectoryString = executablePath.parent_path().string();
-      strncpy(Base_directory, baseDirectoryString.c_str(), sizeof(Base_directory) - 1);
-      Base_directory[sizeof(Base_directory) - 1] = '\0';
-      LOG_INFO << "Using working directory of " << Base_directory;
+      writable_base_directory = executablePath.parent_path();
+      LOG_INFO << "Using working directory of " << writable_base_directory.u8string().c_str();
     }
   } else {
-    ddio_GetWorkingDir(Base_directory, sizeof(Base_directory));
+    writable_base_directory = std::filesystem::current_path();
+  }
+
+  ddio_SetWorkingDir(writable_base_directory.u8string().c_str());
+  initial_base_directories.insert(initial_base_directories.begin(), writable_base_directory);
```

Comment:
Use `.push_back()` on std::vector filling, this is more fast than `.insert()`.

Reply:
OK. I changed it so that if fills `Base_directories` with `.push_back()` and then iterates over `Base_directories` backwards instead of forwards.